.github/workflows/prerelease.yml

```diff
@@ -0,0 +1,99 @@
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: pip install -r requirements-dev.txt
+      - name: Installing Avocado in develop mode
+        run: python3 setup.py develop --user
+      - name: Run make check
+        run: make check
+
+  avocado-pre-release-job:
+    name: Avocado pre-release job
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install avocado
+        run: |
+          sudo python3 -m pip install -r requirements-dev.txt
+          sudo python3 setup.py develop
+      - name: Avocado pre-release job
+        run: ./selftests/pre_release/jobs/pre_release.py
+
+
+  # Re-implements ./selftests/pre_release/tests/check-copr-rpm-version.sh
+  check-rpm-available-copr-job:
+    name: Check the right RPM packages are available in COPR
+    runs-on: ubuntu-latest
+    env:
+      FEDORA_VERSION: '34'
+    container:
+      image: fedora:34
+    steps:
+      - name: install repos and packages
+        run: |
+          dnf -y install git
+          dnf -y module disable avocado
+          dnf -y install 'dnf-command(copr)'
+          dnf -y copr enable @avocado/avocado-latest
+      - uses: actions/checkout@v2
+        with:
+          ref: 'master'
+      - name: Check the right RPM packages are available
+        run: |
+          VERSION=$(python3 setup.py --version 2>/dev/null)
+          RPM_RELEASE_NUMBER=$(grep -E '^Release:\s([0-9]+)' python-avocado.spec | sed -E 's/Release:\s([0-9]+).*/\1/')
+          COMMIT_DATE=$(git log -n 1 --pretty='format:%cd' --date='format:%Y%m%d')
+          SHORT_COMMIT=$(git rev-parse --short=9 master)
+          dnf -y install python3-avocado-${VERSION}-${RPM_RELEASE_NUMBER}.${COMMIT_DATE}git${SHORT_COMMIT}.fc${{env.FEDORA_VERSION}}
+
+
+  avocado-deployment-job:
+    name: Avocado deployment
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:34
+      volumes:
+        - /tmp/artifact:/tmp/artifact
+    env:
+      GIT_URL: 'git://github.com/avocado-framework/avocado'
+      INVENTORY: 'selftests/deployment/inventory'
+      PLAYBOOK: 'selftests/deployment/deployment.yml'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install Ansible
+        run: dnf install -y git ansible
+      - name: Run Ansible playbook
+        run:  ansible-pull -v -U ${{env.GIT_URL}} -i ${{env.INVENTORY}} -c local ${{env.PLAYBOOK}} -e "method=copr"
+      - name: Create artifact directory
+        run: mkdir -p /tmp/artifact
+      - name: Run test
+        run: avocado run /usr/bin/true --html /tmp/artifact/report.html
+      - name: Save artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: avocado-html
+          path: /tmp/artifact
```

Comment:
By reading your cover-letter, iiuc, the idea of this artifact is for doing a manual check after the job finishes. Is this really needed or could we automate this? I mean, I have a feeling that we are not going to look at this artifact at all.

Reply:
I'm trying to make things easier for this step https://github.com/avocado-framework/avocado/blob/master/examples/testplans/release/pre.json#L28
If you were ignoring this step in the pre-release checks, then we can indeed not to store the artifact.

Reply:
We are not ignoring it, my point is that maybe this is the right time to automate this check:

> "Verify that all the links such as `job-YYYY-MM-DD...` (under `Results Dir`), `1-/**/avocado/tests/**/examples/tests/passtest.py:PassTest.test` (under `Test ID`) and `debug.log` point to valid locations."

And if we keep that we way, we have to change the pre release testplan to make sure that we have instructions there, saying: "Double check the HTML artifact generated at job X.."

Reply:
I think we will still need some visual check on the HTML report to be sure everything is right and that the formatting is working properly.

Reply:
Indeed, unless we have a proper selenium-like automated test to check the HTML, we still need to do some visual checks.

Reply:
IMO, a simple regex finding some patterns there could be enough. But I agree that we can delivery this in a different PR. But would be nice to have at least the testplan changed.

Reply:
ok, we can have some automatic checks indeed (and I wanted to look at that eventually).